Sources/NostrSDK/Events/TextNoteEvent.swift

```diff
@@ -37,87 +37,25 @@ public final class TextNoteEvent: NostrEvent, CustomEmojiInterpreting {
     }
 
     /// Pubkeys mentioned in the note content.
+    @available(*, deprecated, message: "Deprecated in favor of referencedPubkeys. Mention is an overloaded term from NIP-10.")
     public var mentionedPubkeys: [String] {
-        allValues(forTagName: .pubkey)
+        referencedPubkeys
     }
-    
-    /// Events mentioned in the note content.
-    public var mentionedEventIds: [String] {
-        allValues(forTagName: .event)
-    }
-
-    /// The ``EventTag`` that denotes the reply event being responded to.
-    /// This event tag may be the same as ``rootEventTag`` if this note is a direct reply to the root of a thread.
-    public var replyEventTag: EventTag? {
-        let eventTags = tags.compactMap { EventTag(tag: $0) }
-
-        // Return the first event tag with a reply marker if it exists.
-        if let reply = eventTags.first(where: { $0.marker == .reply }) {
-            return reply
-        }
-
-        // A direct reply to the root of a thread should have a single marked event tag of type "root".
-        if let root = eventTags.first(where: { $0.marker == .root }) {
-            return root
-        }
-
-        // If there are no reply or root event markers, and there is at least one event tag with a marker,
-        // then we can make a reasonable assumption that the client that created the event does not use
-        // deprecated positional event tags, so there is no reply event tag.
-        guard eventTags.allSatisfy({ $0.marker == nil }) else {
-            return nil
-        }
 
-        // Otherwise, NIP-10 states that the last event tag is the one being responded to.
-        return eventTags.last
+    /// Pubkeys mentioned in the note content.
+    public var referencedPubkeys: [String] {
+        allValues(forTagName: .pubkey)
     }
 
-    /// The ``EventTag`` that denotes the root event of the thread being responded to.
-    public var rootEventTag: EventTag? {
-        let eventTags = tags.compactMap { EventTag(tag: $0) }
-
-        // Return the first event tag with a root marker if it exists.
-        if let root = eventTags.first(where: { $0.marker == .root }) {
-            return root
-        }
-
-        // If there are no root event markers, and there is at least one event tag with a marker,
-        // then we can make a reasonable assumption that the client that created the event does not use
-        // deprecated positional event tags, so there is no root event tag.
-        guard eventTags.allSatisfy({ $0.marker == nil }) else {
-            return nil
-        }
-
-        // NIP-10 states that the first event tag is the root.
-        return eventTags.first
+    /// Events mentioned in the note content.
+    @available(*, deprecated, message: "Deprecated in favor of referencedEventIds. Mention is an overloaded term from NIP-10.")
```

Comment:
[NIP-01](https://github.com/nostr-protocol/nips/blob/master/01.md#tags) says
```
- The `e` tag, used to refer to an event: `["e", <32-bytes lowercase hex of the id of another event>, <recommended relay URL, optional>]`
```

Since this line uses the word `refer`, I've called the new property `referencedEventIds`.